packages/core/saved-objects/core-saved-objects-migration-server-internal/src/zdt/state/types.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { MigrationLog } from '../../types';
+import type { ControlState } from '../../state_action_machine';
+
+export interface BaseState extends ControlState {
+  readonly retryCount: number;
+  readonly retryDelay: number;
+  readonly logs: MigrationLog[];
+}
```

Comment:
State types are different between the two algorithm (both the `BaseState` and the composition of the `State`) so I don't think we can really re-use anything in this specific part of the code.

Reply:
Yeah. TBH I tried just having the logger in the context initially, and it brings two issues:
1. incompatibilities between helper functions of the two algos
2. making it harder to control when logs are outputted 
So I went back to putting the logs in the state...

So if we were to change that, I think we would need to do it in both implementations at the same time, which kinda closed the question for me (given it's not the same amount of work at all)